packages/core/src/submodules/schema/schemas/NormalizedSchema.ts

```diff
@@ -83,34 +80,30 @@ export class NormalizedSchema implements INormalizedSchema {
     }
 
     this.name =
-      (typeof this.schema === "object" ? this.schema?.name : void 0) ?? this.memberName ?? this.getSchemaName();
+      (this.schema instanceof Schema ? this.schema.getName?.() : void 0) ?? this.memberName ?? this.getSchemaName();
 
     if (this._isMemberSchema && !memberName) {
-      throw new Error(
-        `@smithy/core/schema - NormalizedSchema member schema ${this.getName(
-          true
-        )} must initialize with memberName argument.`
-      );
+      throw new Error(`@smithy/core/schema - NormalizedSchema member init ${this.getName(true)} missing member name.`);
     }
   }
 
   public static [Symbol.hasInstance](lhs: unknown): lhs is NormalizedSchema {
-    const isPrototype = NormalizedSchema.prototype.isPrototypeOf(lhs as any);
-    if (!isPrototype && typeof lhs === "object" && lhs !== null) {
-      const ns = lhs as NormalizedSchema;
-      return ns.symbol === NormalizedSchema.symbol;
-    }
-    return isPrototype;
+    return Schema[Symbol.hasInstance].bind(this)(lhs);
   }
 
   /**
    * Static constructor that attempts to avoid wrapping a NormalizedSchema within another.
    */
-  public static of(ref: SchemaRef, memberName?: string): NormalizedSchema {
+  public static of(ref: SchemaRef): NormalizedSchema {
```

Comment:
alpha API breaking change: static public ctor no longer allows a member name.

Members must not be initialized on their own. They must be retrieved as part of an aggregate schema (struct, union, list, map, document etc.). 

This simplifies reasoning about the schema data that is present.